src/utils.cpp

```diff
@@ -105,8 +105,9 @@ void disjunctive_partition(const smt::Term & term,
   }
 }
 
-void get_free_symbolic_consts(const smt::Term & term,
-                              smt::UnorderedTermSet & out)
+void get_kind_terms(const smt::Term & term,
```

Comment:
Maybe there's a better name, like `get_matching_terms`? I think it might be nice to expose this to the API actually and document it in the header file. Then the `get_free_*` are just convenient special cases.